test/e2e/queue.go

```diff
@@ -19,14 +19,184 @@ package e2e
 import (
 	"context"
 	"fmt"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"volcano.sh/volcano/pkg/cli/util"
 )
 
 var _ = Describe("Queue E2E Test", func() {
+	It("Queue Command Close And Open With State Check", func() {
+		q1 := "queue-command-close"
+		ctx := initTestContext(options{
+			queues: []string{q1},
+		})
+		defer cleanupTestContext(ctx)
+
+		By("Close queue command check")
+		err := util.CreateQueueCommand(ctx.vcclient, "default", q1, "CloseQueue")
+		if err != nil {
+			Expect(err).NotTo(HaveOccurred(), "Error send close queue command")
+		}
+
+		err = waitQueueStatus(func() (bool, error) {
+			queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+			return queue.Status.State == "Closed", nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Wait for closed queue %s failed", q1)
+
+		By("Open queue command check")
+		err = util.CreateQueueCommand(ctx.vcclient, "default", q1, "OpenQueue")
+		if err != nil {
+			Expect(err).NotTo(HaveOccurred(), "Error send open queue command")
+		}
+
+		err = waitQueueStatus(func() (bool, error) {
+			queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+			return queue.Status.State == "Open", nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Wait for reopen queue %s failed", q1)
+
+	})
+
+	Describe("Queue Job Status Transaction", func() {
+		var ctx *testContext
+		var q1 string
+		var podNamespace string
+		var rep int32
+		var firstJobName string
+		By("Prepare 3 job")
+		BeforeEach(func() {
+
+			q1 = "queue-jobs-status-transaction"
+			ctx = initTestContext(options{
+				queues: []string{q1},
+			})
+			podNamespace = ctx.namespace
+			slot := oneCPU
+			rep = clusterSize(ctx, slot)
+
+			if rep < 4 {
```

Comment:
This not friendly, should not assume where to run this case.  We can skip this case if the resource can not satisfy

Reply:
Yeah, but i think when resource is not satisfy, we should remind the tester, then he(she) can skip it.
Slot is too big here, i was intended to make the slot half of cpu, and then many laptop can support the case.

Reply:
The Errorf was rewritten to more friendly :)